### DISCUSSION TITLE
Generate Typescript Types on supabase-js v2 (non-local)

### DISCUSSION THREAD

**Discussion by hjaber** [supabase/supabase#8863] [Questions]
What are the proper steps to generate types for a cloud supabase instance? 

Following the CLI [docs](https://supabase.com/docs/reference/javascript/next/typescript-support) and using the --db-url flag instead of --local:

`supabase gen types typescript --db-url postgresql://postgres:[YOUR-PASSWORD]@db.[YOUR-PROEJCT-ID].supabase.co:5432/postgres > lib/database.types.ts`

returns a blank database.ts document or an error stating I'm not running docker which I'm not.

Trying to migrate the V1 method `.from<Message>('messages')` which I really loved to the more robust V2 method of generating types

`supabase cli v1.4.3`
`supabase/supabase-js v2.0.0-rc.9`

**Comment by alex-galey** [comment]
I guess you should be able to run docker, just generated them with the same command now and the execution started with `Pulling docker image...`

Not sure it will solve the problem, but have you `supabase init` or `supabase start` yet ?

**Reply by hjaber** [reply]
> I guess you should be able to run docker, just generated them with the same command now and the execution started with `Pulling docker image...`
> 
> Not sure it will solve the problem, but have you `supabase init` or `supabase start` yet ?

I tried using supabase init/start but they tried to run docker. 

I thought it was possible to run generate types without docker but it doesn't appear so, I'll just go with running docker.
